main.go

```diff
@@ -15,68 +15,121 @@ package main
 
 import (
 	"context"
-	"flag"
+	"fmt"
+	"os"
 
 	"github.com/mendersoftware/go-lib-micro/log"
+	"github.com/urfave/cli"
 
 	"github.com/mendersoftware/deviceauth/config"
 	"github.com/mendersoftware/deviceauth/store/mongo"
 )
 
+const (
+	CliAutomigrate = "automigrate"
+)
+
 func main() {
 	var configPath string
-	var printVersion bool
-	var devSetup bool
 	var debug bool
 
-	flag.StringVar(&configPath, "config",
-		"",
-		"Configuration file path. Supports JSON, TOML, YAML and HCL formatted configs.")
-	flag.BoolVar(&printVersion, "version",
-		false, "Show version")
-	flag.BoolVar(&devSetup, "dev",
-		false, "Use development setup")
-	flag.BoolVar(&debug, "debug",
-		false, "Enable debug logging")
+	app := cli.NewApp()
+	app.Usage = "Tenant Administration Service"
+	app.Version = CreateVersionString()
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:        "config",
+			Usage:       "Configuration `FILE`. Supports JSON, TOML, YAML and HCL formatted configs.",
+			Destination: &configPath,
+		},
+		cli.BoolFlag{
+			Name:  "dev",
+			Usage: "Use development setup",
+		},
+		cli.BoolFlag{
+			Name:        "debug",
+			Usage:       "Enable debug logging",
+			Destination: &debug,
+		},
+	}
 
-	flag.Parse()
+	app.Commands = []cli.Command{
+		{
+			Name:  "server",
+			Usage: "Run the service as a server",
+			Flags: []cli.Flag{
+				cli.BoolFlag{
+					Name:  CliAutomigrate,
```

Comment:
is there anything special about "automigate" that it deserves a const, while "config", "dev" and "debug" do not?